test/images/agnhost/crd-conversion-webhook/converter/framework.go

```diff
@@ -106,18 +132,58 @@ func serve(w http.ResponseWriter, r *http.Request, convert convertFunc) {
 	}
 
 	klog.V(2).Infof("handling request: %v", body)
-	convertReview := v1beta1.ConversionReview{}
-	if _, _, err := serializer.Decode(body, nil, &convertReview); err != nil {
+	// default to v1beta1 for backward compatibility with v1beta1 conversion review requests
+	defaultGVK := v1beta1.SchemeGroupVersion.WithKind("ConversionReview")
+
+	obj, gvk, err := serializer.Decode(body, &defaultGVK, nil)
+	if err != nil {
+		msg := fmt.Sprintf("failed to deserialize body (%v) with error %v", string(body), err)
 		klog.Error(err)
-		convertReview.Response = conversionResponseFailureWithMessagef("failed to deserialize body (%v) with error %v", string(body), err)
-	} else {
-		convertReview.Response = doConversion(convertReview.Request, convert)
-		convertReview.Response.UID = convertReview.Request.UID
+		http.Error(w, msg, http.StatusBadRequest)
+		return
 	}
-	klog.V(2).Info(fmt.Sprintf("sending response: %v", convertReview.Response))
+	if gvk.Empty() {
```

Comment:
the webhook should never see an empty GVK from the server... did you encounter that?

Reply:
I did. It happened when I ran the e2e tests currently on master, which send v1beta1 ConversionReviews. I had expected a apiVersion and kind matching v1beta1 but the fields were missing. I don't think it occurs with the e2e tests upgraded to v1 of ConversionReview.

This will make migrating to v1 Conversion review more difficult for users. I'll create a repro case.

Reply:
I think this is a bug in the webhook's decoding code, not in what the server sends. I checked in 1.15 and the server is definitely sending `{kind:"ConversionReview", apiVersion:"apiextensions.k8s.io/v1beta1",...}` with webhook requests

Reply:
as this will be used in conformance, we want the webhook to be strict about what it accepts from the server, so require the request from the server to set apiVersion and kind to a recognized value.

Reply:
That makes sense. I've removed the check and verified that the v1 e2e tests pass against the updated image.

Reply:
I tried to reproduce this and was unable to. It works correctly both with v1beta1/v1 CRDs and v1beta1/v1 ConversionReview without the check. Maybe it was only a problem before I added types to scheme? (https://github.com/kubernetes/kubernetes/pull/81314/commits/ffde8b4bbcb7f9468af553ef22e736d9194736fc#diff-183fedc9a436d4e6b280571e0fcc75ffR208). Either way, GVKs are getting sent in all cases.